libexec/copier/copier.c

```diff
@@ -698,7 +719,14 @@ main(int argc, char *argv[])
 	if (getuid() == 0 || geteuid() == 0)
 		WARNING("running as root");
 
+	signal(SIGINT, signal_handler);
+	signal(SIGTERM, signal_handler);
+
 	if (tsdfx_copier(argv[0], argv[1], maxsize) != 0)
-		exit(1);
-	exit(0);
+		retval = 1;
+
+	signal(SIGTERM, SIG_DFL);
```

Comment:
It adds complexity (introducing a temporary variable to hold the exit code) solely for the purpose of resetting the signal handler immediately before exiting, which was no real effect except that there is a one-in-a-zillion chance that the copier may catch a `SIGINT` or `SIGTERM` after the two `signal(2)` calls but before the `exit(2)` call, in which case the parent will be informed that we died (i.e. failed) when in fact it is entirely possible that we completed successfully before catching the signal.